src/orchard/info.clj

```diff
@@ -1,35 +1,169 @@
+
+(defn clj-meta
+  [{:keys [dialect ns sym computed-ns unqualified-sym]}]
+  {:pre [(= dialect :clj)]}
+  (let [ns (or ns computed-ns)]
+    (or
+     ;; it's a special (special-symbol?)
+     (m/special-sym-meta sym)
+     ;; it's an unqualified sym for an aliased var
+     (some-> ns (m/resolve-var unqualified-sym) (m/var-meta))
+     ;; it's a var
+     (some-> ns (m/resolve-var sym) (m/var-meta))
+     ;; sym is an alias for another ns
+     (some-> ns (m/resolve-aliases) (get sym) (m/ns-meta))
+     ;; We use :unqualified-sym *exclusively* here because because our :ns is
+     ;; too ambiguous.
+     ;;
+     ;; Observe the incorrect behavior (should return nil, there is a test):
+     ;;
+     ;;   (info '{:ns clojure.core :sym non-existing}) ;;=> {:author "Rich Hickey" :ns clojure.core ...}
+     ;;
+     (some-> (find-ns unqualified-sym) (m/ns-meta))
+     ;; it's a Java class/member symbol...or nil
+     (some-> ns (java/resolve-symbol sym)))))
+
+(defn cljs-meta
+  [{:keys [dialect ns sym env context-ns unqualified-sym]}]
+  {:pre [(= dialect :cljs)]}
+  (let [context-ns (or context-ns ns)]
+    (or
+     ;; a special symbol - always use :unqualified-sym
+     (some-> (cljs-ana/special-meta env unqualified-sym)
+             (cljs-meta/normalize-var-meta))
+     ;; an NS
+     (some->> (cljs-ana/find-ns env sym)
+              (normalize-ns-meta))
+     ;; ns alias
+     (some->> (cljs-ana/ns-alias env sym context-ns)
+              (cljs-ana/find-ns env)
+              (normalize-ns-meta))
+     ;; macro ns
+     (some->> (find-ns unqualified-sym)
+              (cljs-meta/normalize-macro-ns env))
+
+     ;; macro ns alias
+     (some->> (cljs-meta/aliased-macro-var env sym context-ns)
+              (cljs-meta/normalize-macro-ns env))
+     ;; referred var
+     (some->> (get (cljs-ana/referred-vars env context-ns) sym)
+              (cljs-ana/find-symbol-meta env)
+              (cljs-meta/normalize-var-meta))
+     ;; referred macro
+     (some->> (cljs-meta/referred-macro-meta env sym context-ns)
+              (cljs-meta/normalize-macro-meta))
+     ;; scoped var
+     (some->> (cljs-meta/scoped-var-meta env sym context-ns)
+              (cljs-meta/normalize-var-meta))
+     ;; scoped macro
+     (some->> (cljs-meta/scoped-macro-meta env sym context-ns)
+              (cljs-meta/normalize-macro-meta))
+     ;; var in cljs.core
+     (some->> (get (cljs-ana/core-vars env context-ns) sym)
+              (cljs-ana/var-meta)
+              (cljs-meta/normalize-var-meta))
+     ;; macro in cljs.core
+     (some->> (cljs-meta/scoped-macro-meta env sym 'cljs.core)
+              (cljs-meta/normalize-macro-meta)))))
+
 (def see-also-data
   (edn/read-string (slurp (io/resource "see-also.edn"))))
 
-(defn see-also
-  [ns sym]
-  (let [var-key (str ns "/" sym)]
-    (->> (get see-also-data var-key)
-         (filter (comp resolve u/as-sym)))))
+(defn see-also*
+  [{:keys [dialect unqualified-sym] :as m}]
+  (when-not (= :cljs dialect)
```

Comment:
In another patch I am planning to change `see-also.edn` so that we can support ClojureScript as well.